packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -138,44 +179,64 @@ contract Bridge is EssentialContract, IBridge {
         sameChain(message.srcChainId)
     {
         bytes32 msgHash = hashMessage(message);
-        if (isMessageRecalled[msgHash]) revert B_RECALLED_ALREADY();
 
-        address signalService = resolve("signal_service", false);
+        if (messageStatus[msgHash] != Status.NEW) revert B_STATUS_MISMATCH();
 
-        if (!ISignalService(signalService).isSignalSent(address(this), msgHash)) {
-            revert B_MESSAGE_NOT_SENT();
-        }
+        uint64 receivedAt = proofReceipt[msgHash].receivedAt;
+        bool isMessageNotProven = receivedAt == 0;
```

Comment:
[Double negations are not not bad](https://cleankotlin.nl/blog/double-negations)

Just giving the reason why I changed it to the positive case when I did the renaming, I think for most people easier to think about the positive case instead of a name being the negative case.

Reply:
yeah i was searching for the `if( ! xxx)` too :), tho no strong opinion on that.

Reply:
> [Double negations are not not bad](https://cleankotlin.nl/blog/double-negations)
> 
> Just giving the reason why I changed it to the positive case when I did the renaming, I think for most people easier to think about the positive case instead of a name being the negative case.

Changed back since you prefer the original naming.